### ISSUE TITLE
containElementSatisfying dispatch incorrect on CI

### ISSUE THREAD

**Issue by AnthonyMillerSF** [Quick/Nimble#755]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/master/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Using `containElementSatisfying` in swift test using `XCTest` (not `QuickSpec`). Tests compile and run in Xcode 11.4 on my local dev machine. Pushed PR up and our Jenkins CI build started to build.

### What did you expect to happen?

Jenkins CI build to compile and pass test (as occurred on local dev machine)

### What actually happened instead?

Jenkins CI running on Xcode 11.3.1 failed to compile with error
`cannot convert value of type 'NMBMatcher' to expected argument type 'Predicate<[MyProtocol]>'`. 

I think it's dispatching to the Obj-C version of the matcher on my CI machine for some reason.

### Environment

- Quick: N/A
- Nimble: *8.0.6*
- Xcode Version: *1.3.1* (Open Xcode; In menubar: Xcode > About Xcode)
- Swift Version: *Xcode Default* (Open Xcode Preferences; Components > Toolchains. If none, use `Xcode Default`.)

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

- Cocoapods: *1.9.1* (Use `pod --version` in Terminal)

### Code Sample:

```swift
protocol MyProtocol { }

class MyClassA: MyProtocol {
    let name: String?
    init(name: String? { self.name = name }
}

class MyClassB: MyProtocol { }

func test__example() {
    let array: [MyProtocol] = [MyClassA(name: "A"), MyClassB(), MyClassB()]

    expect(array)
            .to(containElementSatisfying {
                guard let actual = $0 as? MyClassA else { return false }
                expect(actual.name).to(equal("A"))
                return true
            })
}
```

**Comment by ikesyo** [comment]
I can't reproduce it locally and the sample code doesn't compile with an error `Class 'MyClassA' has no initializers`.

:thin

**Comment by ikesyo** [comment]
Confirmed that the error occurs on Xcode 10.1/10.3/11.0/11.3 but doesn't on 11.4. So looks like the miscompilation is fixed on Swift 5.2 compiler.

**Comment by ikesyo** [comment]
I realized that the problem is you are using trailing closure syntax but the predicate doesn't support trailing closure because the given predicate block is not the last argument: `public func containElementSatisfying<S: Sequence, T>(_ predicate: @escaping ((T) -> Bool), _ predicateDescription: String = "") -> Predicate<S> where S.Iterator.Element == T`.

See the test code: https://github.com/Quick/Nimble/blob/e55e9582fb403daab0e1d89477320bc86a9718c3/Tests/NimbleTests/Matchers/ContainElementSatisfyingTest.swift#L7-L10

```swift
    expect(array)
            .to(containElementSatisfying({
                guard let actual = $0 as? MyClassA else { return false }
                expect(actual.name).to(equal("A"))
                return true
            }))
```

did work.

**Comment by ikesyo** [comment]
ref: https://github.com/apple/swift/commit/88c477b08d697a3c03ff2ec0bcad64391e2ad031

**Comment by AnthonyMillerSF** [comment]
Hmmm... Thanks so much @ikesyo . I'm pretty sure I tried not using trailing closure syntax, but I could be wrong.
I ended up with the hacky work around of using my own `predicateDescription` parameter, since the Obj-C version doesn't have that param, it forced it to dispatch properly.

Hopefully we will be upgrading our Jenkins boxes to Xcode 11.4 soon, but in the mean time, I've got a workaround.

Appreciate the help!
